craft_application/services/project.py

```diff
@@ -271,8 +274,40 @@ def get_platforms(self) -> dict[str, craft_platforms.PlatformDict]:
                 exc,
                 file_name=self.project_file_name,
             ) from None
+        self._validate_multi_base(self.__platforms)
         return self.__platforms
 
+    def _validate_multi_base(
+        self, platforms: dict[str, craft_platforms.PlatformDict]
+    ) -> None:
+        """Ensure that the given platforms are not multi-base.
+
+        :param platforms: The platforms mapping to ensure is multi-base.
```

Comment:
I feel slightly misled by the function name and docstrings, since it doesn''t validate that multi-base notation is used correctly if the app supports multi-base notation.  But I don't have a better suggestion.

Reply:
I originally had `_ensure_not_multi_base` but I thought this would be nice for overriding if we do have a multi-base-capable app. Maybe I should be more explicit about the intent of overriding this?

Reply:
WDYT?

Reply:
Yeah, I think that extra docstring explains the intention well, thanks.